bundler/lib/bundler/resolver.rb

```diff
@@ -79,8 +80,7 @@ def setup_solver
     def solve_versions(root:, logger:)
       solver = PubGrub::VersionSolver.new(source: self, root: root, logger: logger)
       result = solver.solve
-      resolved_specs = result.map {|package, version| version.to_specs(package) }.flatten
-      resolved_specs |= @base.specs_compatible_with(SpecSet.new(resolved_specs))
+      result.map {|package, version| version.to_specs(package, @most_specific_locked_platform) }.flatten
```

Comment:
`flat_map` instead of `map.flatten` ?

Reply:
Sure, we can probably add a performance RuboCop check for this!

Reply:
I'll add this separately in https://github.com/rubygems/rubygems/pull/8261.